src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -634,6 +589,28 @@ public JsonPropertyInfo CreateJsonPropertyInfo(Type propertyType, string name)
             return propertyInfo;
         }
 
+        [MemberNotNull(nameof(_properties))]
+        private void PopulatePropertyList()
+        {
+            if (!_isConfigured)
```

Comment:
`Debug.Assert(_properties == null);` inside the lock
also consider doing that check here and renaming to `EnsurePropertyListPopulated`

Reply:
It's possible for the properties field to be non-null when this method is being called.